src/Microsoft.Sbom.Parsers.Spdx30SbomParser/Generator.cs

```diff
@@ -615,5 +618,44 @@ private PackageVerificationCode GetPackageVerificationCode(IInternalMetadataProv
 
     public ManifestInfo RegisterManifest() => Constants.Spdx30ManifestInfo;
 
-    IDictionary<string, object> IManifestGenerator.GetMetadataDictionary(IInternalMetadataProvider internalMetadataProvider) => throw new NotSupportedException();
+    public IDictionary<string, object> GetMetadataDictionary(IInternalMetadataProvider internalMetadataProvider)
+    {
+        if (internalMetadataProvider is null)
```

Comment:
This code has a lot of overlap with the code in the `GenerateJsonDocument` method (starts on line 331), with some funny differences--344 uses string accretion, 646 uses interpolation. Is there an opportunity to unify some of this code?